src/functions.jl

```diff
@@ -369,3 +369,59 @@ Return a new quadratic function with a shallow copy of the terms and constant(s)
 from `func`.
 """
 Base.copy(func::F) where {F <: Union{ScalarQuadraticFunction, VectorQuadraticFunction}} = F(copy(func.affine_terms), copy(func.quadratic_terms), copy(_constant(func)))
+
+# Define shortcuts for
+# SingleVariable -> ScalarAffineFunction
+function ScalarAffineFunction{T}(f::SingleVariable) where T
```

Comment:
Is it still Julia convention (post-1.0) to add constructor methods for conversions like this?

Reply:
If I understand correctly, since constructors are the low level functions  for building objects, convert should redirect to constructors not the opposite